In [6]:
%pip install folium pandas OSMnx networkx scikit-learn matplotlib numpy

Note: you may need to restart the kernel to use updated packages.


In [7]:
import solution_reader
import os
import pandas as pd
import numpy as np

In [8]:
# Files

solution_directory = "./../output/mini_200_bank_90s"
best_known_file = "./../best_known/best_known_LL01_100_200.txt"

# Best known dataframe
best_known_df = solution_reader.read_best_known(best_known_file)

# Solution dataframe
solutions_list = []
files = os.listdir(solution_directory)
for solution_file in files :
    solution = solution_reader.PDPTWCompleteSolution(solution_directory + "/" + solution_file)
    
    solution_dict = {
    "Instance": solution.get_instance(),
    "Vehicles": solution.get_vehicles(),
    "Distance": solution.get_cost(),
    "Time": solution.get_time(),
    "Reference": solution.get_reference(),
    "Date": solution.get_date(),
    "Time Best": solution.get_best_time_fleet(),
    "Iteration Best": solution.get_best_iteration_fleet()
    }

    solutions_list.append(solution_dict)

solutions_df = pd.DataFrame(solutions_list)


best_known_df = best_known_df[best_known_df["Instance"].isin(solutions_df["Instance"])]
solutions_df.sort_values(by=["Instance"], inplace=True)
best_known_df.sort_values(by=["Instance"], inplace=True)

solutions_df.reset_index(drop=True, inplace=True)
best_known_df.reset_index(drop=True, inplace=True)

print(best_known_df)
print(solutions_df)

    Instance  Vehicles  Distance Reference       Date
0    LC1_2_1        20   2704.57    Li&Lim       2001
1    LC1_2_2        19   2764.56    Li&Lim       2001
2    LC1_2_3        17   3127.78         H  15-jul-16
3    LC2_2_1         6   1931.44    Li&Lim       2001
4    LC2_2_2         6   1881.40    Li&Lim       2001
5    LC2_2_3         6   1844.33  SAM::OPT  15-apr-03
6    LR1_2_1        20   4819.12    Li&Lim       2001
7    LR1_2_2        17   4621.21        RP  08-jul-03
8    LR1_2_3        14   4402.38       CLS  27-feb-16
9    LR2_2_1         5   4073.10  SAM::OPT  09-dec-03
10   LR2_2_2         4   3796.00  SAM::OPT  14-feb-03
11   LR2_2_3         4   3098.36        RP  08-jul-03
12  LRC1_2_1        19   3606.06  SAM::OPT  14-dec-03
13  LRC1_2_2        15   3671.02       EOE  19-nov-15
14  LRC1_2_3        13   3154.92       CLS  27-feb-16
15  LRC2_2_1         6   3595.18         K  14-apr-15
16  LRC2_2_2         5   3158.25         H  13-jul-16
17  LRC2_2_3         4   288

/home/a24jacqb/Documents/Code/pdptw-main/python/solution_reader.py:18: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
  df = pd.read_csv(file, delim_whitespace=True, names=["Instance", "Vehicles", "Distance", "Reference", "Date"], skiprows=1)


In [9]:
sum_vehicles_best_known = np.sum(best_known_df["Vehicles"])
sum_vehicles_solution = np.sum(solutions_df["Vehicles"])
sum_time = np.sum(solutions_df["Time"])

mean_vehicles_best_known = np.mean(best_known_df["Vehicles"])
mean_vehicles_solution = np.mean(solutions_df["Vehicles"])

mean_best_time = np.mean(solutions_df["Time Best"])
mean_best_iteration = np.mean(solutions_df["Iteration Best"])

# Percentage error is calculated as (MAE / Actual value) * 100
percentage_error_vehicles = ((mean_vehicles_solution - mean_vehicles_best_known) / mean_vehicles_best_known) * 100



print(f"Vehicle difference: {sum_vehicles_solution - sum_vehicles_best_known}")
print(f"Percentage Error (vehicles): {percentage_error_vehicles:.2f}%")
print(f"Average time to find the best : {mean_best_time:.2f}s")
print(f"Average iteration to find the best : {mean_best_iteration:.2f}")

Vehicle difference: 18
Percentage Error (vehicles): 9.00%
Average time to find the best : 0.06s
Average iteration to find the best : 377.39


mini_200_bank_90s

- Vehicle difference: 18
- Percentage Error (vehicles): 9.00%
- Average time to find the best : 0.06s
- Average iteration to find the best : 377.39

mini_200_90s

- Vehicle difference: 6
- Percentage Error (vehicles): 3.00%
- Average time to find the best : 2.72s
- Average iteration to find the best : 11647.28

Bank focus ne semble pas fonctionner sur les grandes instances !

Résultats mini_100_30s

- Vehicle difference: 2
- Percentage Error (vehicles): 1.39%
- Average time to find the best : 1.10s
- Average iteration to find the best : 11322.11

(lc103, lr202)

Résultats mini_100_bank_30s

- Vehicle difference: 2
- Percentage Error (vehicles): 1.39%
- Average time to find the best : 0.26s
- Average iteration to find the best : 1965.44

(lrc201, lrc202)

Résultats mini_100_rand_30s

- Vehicle difference: 6
- Percentage Error (vehicles): 4.17%
- Average time to find the best : 0.05s
- Average iteration to find the best : 413.50

Résultats mini_100_rand_bank_30s

- Vehicle difference: 5
- Percentage Error (vehicles): 3.47%
- Average time to find the best : 0.05s
- Average iteration to find the best : 366.56

Conclusion sur taille 100 :

- Bank focus string removal mieux que string removal
- suppression intelligente des routes mieux que supressions aléatoire